flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointPlanCalculator.java

```diff
@@ -30,6 +30,8 @@
      * Calculates the plan of the next checkpoint.
      *
      * @return The result plan.
+     * @param isUnalignedCheckpoint {@code true} if plan should be calculated for unaligned
+     *     checkpoint.
      */
-    CompletableFuture<CheckpointPlan> calculateCheckpointPlan();
+    CompletableFuture<CheckpointPlan> calculateCheckpointPlan(boolean isUnalignedCheckpoint);
```

Comment:
Perhaps, it makes sense to make this flag as the field of DefaultCheckpointPlanCalculator because as I understand, it is impossible to change the checkpoint configuration on runtime.